lara-typescript/src/interactive-api-client/api.ts

```diff
@@ -396,14 +396,13 @@ type WriteAttachmentParams = Omit<IWriteAttachmentRequest, "requestId" | "operat
 export const writeAttachment = (params: WriteAttachmentParams): Promise<Response> => {
   return new Promise<Response>((resolve, reject) => {
     const client = getClient();
-    const { content, contentType, options, ...others } = params;
+    const { content, contentType, options = {}, ...others } = params;
     const request: IAttachmentUrlRequest = { ...others, operation: "write", requestId: client.getNextRequestId() };
     client.addListener("attachmentUrl", async (response: IAttachmentUrlResponse) => {
```

Comment:
Because this is going close around the content variable which might be large. Did you check if the listener is automatically removed from the client after the message is received? Otherwise I think this will cause a memory leak.

Reply:
Yes, it is automatically removed [here](https://github.com/concord-consortium/lara/blob/0f33f2d2fc8e0fd15207c315f5e66a2cb1cd6a5c/lara-typescript/src/interactive-api-client/client.ts#L105-L107) for any message that contains a `requestId`.